api/envoy/config/core/v3/protocol.proto

```diff
@@ -651,6 +651,13 @@ message Http3ProtocolOptions {
 message SchemeHeaderTransformation {
   oneof transformation {
     // Overwrite any Scheme header with the contents of this string.
```

Comment:
why not have a transformation which is set_to_match_upstream?

Reply:
See resolved conversation above